UnitTests/BraintreeCoreTests/Analytics/BTAnalyticsService_Tests.swift

```diff
@@ -35,19 +35,23 @@ final class BTAnalyticsService_Tests: XCTestCase {
         analyticsService.flushThreshold = 1
         analyticsService.http = mockAnalyticsHTTP
 
-        analyticsService.sendAnalyticsEvent("an.analytics.event")
-
-        // Pause briefly to allow analytics service to dispatch async blocks
-        RunLoop.current.run(until: Date(timeIntervalSinceNow: 0.1))
-
-        XCTAssertEqual(mockAnalyticsHTTP.lastRequestEndpoint, "v1/tracking/batch/events")
+        let expectation = expectation(description: "Sends analytics event")
         
-        let timestamp = parseTimestamp(mockAnalyticsHTTP.lastRequestParameters)!
-        let eventName = parseEventName(mockAnalyticsHTTP.lastRequestParameters)
-        XCTAssertEqual(eventName, "an.analytics.event")
-        XCTAssertGreaterThanOrEqual(timestamp, currentTime)
-        XCTAssertLessThanOrEqual(timestamp, oneSecondLater)
-        validateMetadataParameters(mockAnalyticsHTTP.lastRequestParameters)
+        analyticsService.sendAnalyticsEvent("any.analytics.event") { error in
+            expectation.fulfill()
+        }
```

Comment:
do we need  parameter`error`? maybe we can change it to `_` or validate if the error is different to nil
```suggestion
        analyticsService.sendAnalyticsEvent("any.analytics.event") { error in
            if let error {
               // assert
            } else {
               expectation.fulfill()
            }
        }
```

Reply:
same question for all the events below with error parameter

Reply:
Good catch. I just merged and I can make a change in a follow up PR